mlir/include/mlir/Dialect/Rock/Generator/ConvGenerator.h

```diff
@@ -39,10 +41,10 @@ class Conv2dGenerator {
     std::string filterDataTypeStr;
     std::string inputDataTypeStr;
     std::string outputDataTypeStr;
-    int dilationHeight, dilationWidth;
-    int strideHeight, strideWidth;
-    int paddingHeightLeft, paddingHeightRight;
-    int paddingWidthLeft, paddingWidthRight;
+    SmallVector<int, 5> dilationDims;
```

Comment:
Arbitrary 5.  Krzysztof suggested 4 would be okay, as long as it's generalisable.  Should it be a symbolic constant?

Reply:
1) While we're here, make it `int64_t`
2) I'd argue for 4 here because it's better aligned

Reply:
Making them int64_t had a lot of ripple effects.  See the last commit.